## Comparing the Performance of Injector Wells vs. Producer Wells

In [3]:
import pandas as pd

# Load the data from the CSV file
df = pd.read_csv('C:\\Users\\ihsankoo\\Downloads\\Liang_Cleaned.csv')

To compare the performance of injector wells and producer wells, we can examine key metrics like the total volume of fluid (oil, water, gas) produced or injected over the dataset's time span.

Let's start by aggregating the total fluid volumes for each type of well (injector vs. producer).

In [4]:
# Aggregating the total fluid volumes for injector and producer wells
total_volumes_by_type = df.groupby('Is Injector Well').agg({
    'Oil Production Rate': 'sum',
    'Water Production Rate': 'sum',
    'Gas Production Rate': 'sum',
    'Water Injection Rate': 'sum',
    'Gas Injection Rate': 'sum'
}).rename(index={0: 'Producer', 1: 'Injector'})

total_volumes_by_type


Oil Production Rate  Water Production Rate  \
Is Injector Well                                               
Producer                    134480.99              420096.04   
Injector                         0.00                   0.00   

                  Gas Production Rate  Water Injection Rate  \
Is Injector Well                                              
Producer                            0                  0.00   
Injector                            0             557997.26   

                  Gas Injection Rate  
Is Injector Well                      
Producer                           0  
Injector                           0

Here's the aggregated data comparing the performance of injector wells and producer wells:

#### Producer Wells:

* Oil Production: 134,480.99 units
* Water Production: 420,096.04 units
* Gas Production: 0 units (as previously observed)


#### Injector Wells:

* Water Injection: 557,997.26 units
* Gas Injection: 0 units (as previously observed)


#### From this data:

* We can see that while producer wells have both oil and water production, injector wells only have water injection rates.
* Producer wells do not have any gas production, and injector wells do not have any gas injection, which is consistent with our previous analyses.